Source/WebCore/rendering/RenderThemeIOS.mm

```diff
@@ -725,9 +725,8 @@ static bool renderThemePaintSwitchTrack(OptionSet<ControlStyle::State>, const Re
 {
     // FIXME: Deduplicate sizing with the generic code somehow.
     if (style.width().isAuto() || style.height().isAuto()) {
-        auto size = std::max(style.computedFontSize(), logicalSwitchHeight);
-        style.setLogicalWidth({ size * (logicalSwitchWidth / logicalSwitchHeight), LengthType::Fixed });
-        style.setLogicalHeight({ size, LengthType::Fixed });
+        style.setLogicalWidth({ logicalSwitchWidth * style.effectiveZoom(), LengthType::Fixed });
+        style.setLogicalHeight({ logicalSwitchHeight * style.effectiveZoom(), LengthType::Fixed });
```

Comment:
This seems a little strange. Can you confirm using Web Inspector that the computed width/height are unaffected when using the `zoom` property?

I'm also curious why switch is special over regular checkboxes. Do they already have a similar issue?

Reply:
It seemed strange to me as well! I have now added a test to demonstrate that.

Checkboxes are almost equal to this, except they use min-width/min-height to set this size. It's not clear to me that is better and in fact it seems more complicated for web developers to deal with.

Reply:
We might be able to upstream this test at some point given https://github.com/w3c/csswg-drafts/issues/9398.